independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanGenerator.java

```diff
@@ -1967,7 +2064,9 @@ private void initializeProxy(BeanInfo bean, String baseName, ClassCreator beanCr
     public static ResultHandle getJavaMemberHandle(MethodCreator constructor,
             InjectionPointInfo injectionPoint, ReflectionRegistration reflectionRegistration) {
         ResultHandle javaMemberHandle;
-        if (Kind.FIELD.equals(injectionPoint.getTarget().kind())) {
+        if (injectionPoint.isSynthetic()) {
+            javaMemberHandle = constructor.loadNull();
+        } else if (Kind.FIELD.equals(injectionPoint.getTarget().kind())) {
```

Comment:
Unrelated: can use `injectionPoint.isField()` here, for symmetry with `isSynthetic()`.

Reply:
Good point.